### PR TITLE
Add information on default entryPoint property values

### PR DESCRIPTION
**Pull request by chrisdavidmills**
### Description

In Chrome 121, you can omit the `entryPoint` property values of the compute, fragment, and vertex shader object definitions contained in the configuration objects of [`createComputePipeline()`](https://developer.mozilla.org/en-US/docs/Web/API/GPUDevice/createComputePipeline)/[`createComputePipelineAsync()`](https://developer.mozilla.org/en-US/docs/Web/API/GPUDevice/createComputePipelineAsync) and [`createRenderPipeline()`](https://developer.mozilla.org/en-US/docs/Web/API/GPUDevice/createRenderPipeline)/[`createRenderPipelineAsync()`](https://developer.mozilla.org/en-US/docs/Web/API/GPUDevice/createRenderPipelineAsync), provided your shader code contains a single function that the browser can determine as the default entry point.

This PR adds information and validation entries about default entry points to the above pages.

See https://developer.chrome.com/blog/new-in-webgpu-121#default_entry_points_to_shader_modules for evidence of the addition.

### Related issues and pull requests

Project issue: https://github.com/mdn/content/issues/36354

### CODE REVIEW

File: files/en-us/web/api/gpudevice/createrenderpipeline/index.md [hunk 1]
```diff
@@ -133,8 +133,10 @@ The `fragment` object contains an array of objects, each of which can contain th
     }
     ```
 
-- `entryPoint`
+- `entryPoint` {{optional_inline}}
   - : The name of the function in the `module` that this stage will use to perform its work. The corresponding shader function must have the `@fragment` attribute to be identified as this entry point. See [Entry Point Declaration](https://gpuweb.github.io/gpuweb/wgsl/#entry-point-decl) for more information.
+    > [!NOTE]
+    > You can omit the `entryPoint` property if your shader code contains a single fragment shader entry point function — the browser will use this as the default entry point. If `entryPoint` is omitted and the browser cannot determine a default entry point, a {{domxref("GPUValidationError")}} is generated and the resulting {{domxref("GPUComputePipeline")}} will be invalid.
```
Comment (beaufortfrancois) [inline files/en-us/web/api/gpudevice/createrenderpipeline/index.md:139]: It's not a GPUComputePipeline here but a GPURenderPipeline I believe

File: files/en-us/web/api/gpudevice/createrenderpipeline/index.md [hunk 2]
```diff
@@ -313,8 +315,10 @@ The `vertex` object can contain the following properties:
     }
     ```
 
-- `entryPoint`
+- `entryPoint` {{optional_inline}}
   - : The name of the function in the `module` that this stage will use to perform its work. The corresponding shader function must have the `@vertex` attribute to be identified as this entry point. See [Entry Point Declaration](https://gpuweb.github.io/gpuweb/wgsl/#entry-point-decl) for more information.
+    > [!NOTE]
+    > You can omit the `entryPoint` property if your shader code contains a single vertex shader entry point function — the browser will use this as the default entry point. If `entryPoint` is omitted and the browser cannot determine a default entry point, a {{domxref("GPUValidationError")}} is generated and the resulting {{domxref("GPUComputePipeline")}} will be invalid.
```
Comment (beaufortfrancois) [inline files/en-us/web/api/gpudevice/createrenderpipeline/index.md:321]: Same here. It's not a GPUComputePipeline here but a GPURenderPipeline

File: files/en-us/web/api/gpudevice/createrenderpipelineasync/index.md [hunk 3]
```diff
@@ -49,6 +49,11 @@ A validation error can occur if any of the following are false:
   - `targets.length` is less than or equal to the {{domxref("GPUDevice")}}'s `maxColorAttachments` {{domxref("GPUSupportedLimits", "limit", "", "nocode")}}.
   - For each `target`, `writeMask`'s numeric equivalent is less than 16.
   - If any of the used blend factor operations use the source alpha channel (for example `"src-alpha-saturated"`), the output has an alpha channel (that is, it must be a `vec4`).
+  - If the `entryPoint` property is omitted, the shader code contains a single fragment shader entry point function for the browser to use as the default entry point.
+- For `primitive` objects:
+  - If the `unclippedDepth` property is used, the `depth-clip-control` [feature](/en-US/docs/Web/API/GPUSupportedFeatures) is enabled.
```
Comment (beaufortfrancois) [inline files/en-us/web/api/gpudevice/createrenderpipelineasync/index.md:54]: I believe this line should be part of another PR.

**Review by beaufortfrancois — COMMENTED** [review]
LGTM with nits

File: files/en-us/web/api/gpudevice/createrenderpipeline/index.md (hunk 1 above)
Comment (chrisdavidmills) [reply files/en-us/web/api/gpudevice/createrenderpipeline/index.md]: Ooops, yes! Fixed now!

File: files/en-us/web/api/gpudevice/createrenderpipeline/index.md (hunk 2 above)
Comment (chrisdavidmills) [reply files/en-us/web/api/gpudevice/createrenderpipeline/index.md]: Fixed. Thanks for spotting it.

File: files/en-us/web/api/gpudevice/createrenderpipelineasync/index.md (hunk 3 above)
Comment (chrisdavidmills) [reply files/en-us/web/api/gpudevice/createrenderpipelineasync/index.md:54]: Strictly speaking, yes. I added this info as part of a previous PR, but only added this validation note to the `createRenderPipeline()` page. I noticed it while doing this bit of work, and decided that it wasn't that big a deal to just add it.

File: files/en-us/web/api/gpudevice/createcomputepipeline/index.md [hunk 4]
```diff
@@ -51,8 +51,10 @@ createComputePipeline(descriptor)
             }
             ```
 
-        - `entryPoint`
+        - `entryPoint` {{optional_inline}}
           - : The name of the function in the `module` that this stage will use to perform its work. The corresponding shader function must have the `@compute` attribute to be identified as this entry point. See [Entry Point Declaration](https://gpuweb.github.io/gpuweb/wgsl/#entry-point-decl) for more information.
+            > [!NOTE]
+            > You can omit the `entryPoint` property if your shader code contains a single compute shader entry point function — the browser will use this as the default entry point. If `entryPoint` is omitted and the browser cannot determine a default entry point, a {{domxref("GPUValidationError")}} is generated and the resulting {{domxref("GPUComputePipeline")}} will be invalid.
```
Comment (wbamberg) [inline files/en-us/web/api/gpudevice/createcomputepipeline/index.md:57]: Does "a single compute shader entry point function" mean "a single function with the `@compute` attribute set"? Is it worth saying? It's a little odd (to me) that we use this precise formulation in the previous para, then the more general one here.

File: files/en-us/web/api/gpudevice/createcomputepipeline/index.md [hunk 5]
```diff
@@ -51,8 +51,10 @@ createComputePipeline(descriptor)
             }
             ```
 
-        - `entryPoint`
+        - `entryPoint` {{optional_inline}}
           - : The name of the function in the `module` that this stage will use to perform its work. The corresponding shader function must have the `@compute` attribute to be identified as this entry point. See [Entry Point Declaration](https://gpuweb.github.io/gpuweb/wgsl/#entry-point-decl) for more information.
+            > [!NOTE]
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/api/gpudevice/createcomputepipeline/index.md:56]: No particular reason; I think you are right. I have updated them to normal paras in the next commit.

**Review by wbamberg — APPROVED** [review]
👍 thanks Chris!
